pennylane/devices/experimental/default_qubit_2.py

```diff
@@ -137,6 +138,7 @@ def name(self):
     def __init__(self, seed=None, max_workers=None) -> None:
         super().__init__()
         self._max_workers = max_workers
+        seed = seed or np.random.randint(0, high=10000000)
```

Comment:
does this meet the requirements/expectations of users? eg. if this happens, is that ok?
```python
np.random.seed(42)
dev1 = DefaultQubit2()

np.random.seed(42)
np.random.randint(10000)
dev2 = DefaultQubit2()
```
I guess there's nothing to do because numpy doesn't expose the rng seed, so I'm just looking for your thoughts on it.

Reply:
right, because whatever nonsense happens between seed-setting and device initialization, we're assuming it's the _same_ nonsense in our end-user's workflow on each execution. lgtm